content/en/docs/tasks/security/authorization/authz-ingress/test.sh

```diff
@@ -37,24 +37,35 @@ _set_ingress_environment_variables
 
 _verify_same snip_before_you_begin_3 "$snip_before_you_begin_3_out"
 
-_verify_like snip_before_you_begin_4 "$snip_before_you_begin_4_out"
-
-CLIENT_IP=$(curl "$INGRESS_HOST":"$INGRESS_PORT"/ip -s | grep "origin" | cut -d'"' -f 4)
-
 snip_ipbased_allow_list_and_deny_list_1
 _wait_for_istio authorizationpolicy istio-system ingress-policy
 
-_verify_same snip_ipbased_allow_list_and_deny_list_2 "$snip_ipbased_allow_list_and_deny_list_2_out"
+_verify_same snip_ipbased_allow_list_and_deny_list_3 "$snip_ipbased_allow_list_and_deny_list_3_out"
 
-snip_ipbased_allow_list_and_deny_list_3
+snip_ipbased_allow_list_and_deny_list_2
 _wait_for_istio authorizationpolicy istio-system ingress-policy
 
-_verify_same snip_ipbased_allow_list_and_deny_list_4 "$snip_ipbased_allow_list_and_deny_list_4_out"
+_verify_contains snip_ipbased_allow_list_and_deny_list_4 "remoteIP"
 
 snip_ipbased_allow_list_and_deny_list_5
+
+snip_ipbased_allow_list_and_deny_list_6
 _wait_for_istio authorizationpolicy istio-system ingress-policy
 
-_verify_same snip_ipbased_allow_list_and_deny_list_6 "$snip_ipbased_allow_list_and_deny_list_6_out"
+snip_ipbased_allow_list_and_deny_list_7
+_wait_for_istio authorizationpolicy istio-system ingress-policy
+
+snip_ipbased_allow_list_and_deny_list_8
+
+snip_ipbased_allow_list_and_deny_list_9
+_wait_for_istio authorizationpolicy istio-system ingress-policy
+
+snip_ipbased_allow_list_and_deny_list_10
+_wait_for_istio authorizationpolicy istio-system ingress-policy
+
+snip_ipbased_allow_list_and_deny_list_11
+
+snip_ipbased_allow_list_and_deny_list_12
```

Comment:
Hmm, is there nothing to verify for any of these snips?

Reply:
Great question, the original docs use examples with ipBlocks and assume that the cluster is configured with "externalTrafficPolicy: local" so that the actual original client IP address is available as the packet source address in k8s.  The $CLIENT_IP env var was also originally set with this in mind.  However, with the remoteIpBlocks examples, the way remote.ip gets set is highly dependent on the k8s environment: which load balancer, proxy protocol or X-Forwarded-For header, and meshConfig.

I don't know of a clean way to get the remoteIP for the CLIENT_IP setting, so I left the remoteIP examples without a verify.  The original ipBlocks examples still have the verify though.

Reply:
I think as long as the original ipBlocks field is covered, we're mostly good here as it's just about using two fields in the API, the doc structure, steps and commands are mostly the same (except the environment setup that is kind out of the scope and hard to reproduce in test).

And we also have e2e tests covering both fields in the istio/istio repo.

Reply:
Breaking things up into tabs made it easier to add verifications, so I added some more.  I got the client ip from the envoy debug logs, which feels a little bit like gaming it, but it does work.